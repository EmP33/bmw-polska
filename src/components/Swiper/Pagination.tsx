import React from "react";
// Styles
import { Pagination } from "./Swiper.styles";

interface Props {
  swipeSlide: number;
  handleChangeSlide: (numb: number) => void;
}

const PaginationComponent: React.FC<Props> = ({
  swipeSlide,
  handleChangeSlide,
}) => {
  return (
    <Pagination>
      <label className="container">
        <input
          data-testid="pag1"
          type="radio"
          name="pagination"
          id="pag1"
          onChange={handleChangeSlide.bind(null, 0)}
          checked={swipeSlide === 0}
        />
        <span className="checkmark"></span>
      </label>
      <label className="container">
        <input
          data-testid="pag2"
          type="radio"
          name="pagination"
          id="pag2"
          onChange={handleChangeSlide.bind(null, 1)}
          checked={swipeSlide === 1}
        />
        <span className="checkmark"></span>
      </label>
      <label className="container">
        <input
          data-testid="pag3"
          type="radio"
          name="pagination"
          id="pag3"
          onChange={handleChangeSlide.bind(null, 2)}
          checked={swipeSlide === 2}
        />
        <span className="checkmark"></span>
      </label>
    </Pagination>
  );
};

export default PaginationComponent;
