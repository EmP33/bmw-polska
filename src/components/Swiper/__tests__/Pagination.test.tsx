import Pagination from "../Pagination";
import ReactShallowRenderer from "react-test-renderer/shallow";
import { render, screen, fireEvent } from "@testing-library/react";

describe("<Pagination />", () => {
  it("should render Pagination Component properly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <Pagination swipeSlide={0} handleChangeSlide={jest.fn()} />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("should first pagination radio button be checked by default", () => {
    render(<Pagination swipeSlide={0} handleChangeSlide={jest.fn()} />);
    const inputElement = screen.getByTestId("pag1") as HTMLInputElement;
    expect(inputElement.checked).toEqual(true);
  });

  it("should change swipeSlide value when every radio button is clicked", () => {
    let swipeSlide = 0;
    const handleChange = jest.fn().mockImplementation((numb: number) => {
      swipeSlide = numb;
    });
    render(
      <Pagination swipeSlide={swipeSlide} handleChangeSlide={handleChange} />
    );
    const inputElement1 = screen.getByTestId("pag1") as HTMLInputElement;
    const inputElement2 = screen.getByTestId("pag2") as HTMLInputElement;
    const inputElement3 = screen.getByTestId("pag3") as HTMLInputElement;

    expect(swipeSlide).toBe(0);

    fireEvent.click(inputElement1);
    expect(swipeSlide).toBe(0);

    fireEvent.click(inputElement2);
    expect(swipeSlide).toBe(1);

    fireEvent.click(inputElement3);
    expect(swipeSlide).toBe(2);
  });
});
