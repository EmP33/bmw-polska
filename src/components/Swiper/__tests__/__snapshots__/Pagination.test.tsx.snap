// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render Pagination Component properly 1`] = `
<styled.ul>
  <li>
    <label
      className="container"
    >
      <input
        checked={true}
        data-testid="pag1"
        id="pag1"
        name="pagination"
        onChange={[Function]}
        type="radio"
      />
      <span
        aria-checked={true}
        aria-label="Switch slide"
        className="checkmark"
        role="radio"
      />
    </label>
  </li>
  <li>
    <label
      className="container"
    >
      <input
        checked={false}
        data-testid="pag2"
        id="pag2"
        name="pagination"
        onChange={[Function]}
        type="radio"
      />
      <span
        aria-checked={true}
        aria-label="Switch slide"
        className="checkmark"
        role="radio"
      />
    </label>
  </li>
  <li>
    <label
      className="container"
    >
      <input
        checked={false}
        data-testid="pag3"
        id="pag3"
        name="pagination"
        onChange={[Function]}
        type="radio"
      />
      <span
        aria-checked={true}
        aria-label="Switch slide"
        className="checkmark"
        role="radio"
      />
    </label>
  </li>
</styled.ul>
`;
